src/Microsoft.AspNetCore.Mvc.ApiExplorer/ApiResponseTypeProvider.cs

```diff
@@ -46,7 +46,13 @@ public ICollection<ApiResponseType> GetApiResponseTypes(ControllerActionDescript
                 responseMetadataAttributes = apiConventionResult.ResponseMetadataProviders;
             }
 
-            var apiResponseTypes = GetApiResponseTypes(responseMetadataAttributes, runtimeReturnType);
+            var defaultErrorType = typeof(void);
+            if (action.Properties.TryGetValue(typeof(ApiErrorTypeAttribute), out result))
+            {
+                defaultErrorType = (Type)result;
```

Comment:
Think about storing the attribute itself rather thatn just the type here. It's common when something uses a type as a key, to find a value of that type.